cmd/pulumi.go

```diff
@@ -76,6 +81,15 @@ func NewPulumiCmd() *cobra.Command {
 				}
 			}
 
+			if curVer, err := semver.ParseTolerant(version.Version); err == nil {
+				client := client.NewClient(cloud.DefaultURL(), "")
+				if warn, latest, err := client.CheckIfCLINeedsUpdate(commandContext(), curVer); err == nil && warn {
```

Comment:
FWIW we already send the CLI's version in the `User-Agent` header with API requests. Though I believe we should pass the version explicitly, like you have here, rather than parsing it from the header server-side.
https://github.com/pulumi/pulumi/blob/master/pkg/backend/cloud/client/api.go#L136

Reply:
Yeah, I noticed we did that but then decided I did not want to try to parse that header :-).